commit 575e3207af3cb7ade1a0cf4534b27176cbe09afc (refs/changes/96/5096/5)
Author: Jan Wyszynski <jan.wyszynski@joyent.com>
Date:   2018-12-04T21:20:43+00:00 (10 months ago)
    
    MANTA-4031 remove unnecessary muskie assert from disabled snaplinks check
    MANTA-4032 muskie_deleted_bytes metric should not include failed delete requests

diff --git a/lib/audit.js b/lib/audit.js
index fcfee03..680ba39 100644
--- a/lib/audit.js
+++ b/lib/audit.js
@@ -299,7 +299,11 @@ function auditLogger(options) {
             latency_histogram.observe(obj.latencyToFirstByte, labels);
         }
 
-        if (op === 'DELETE' && req.metadata) {
+        /*
+         * Only count storage freed if the delete was successful. Today, all
+         * Muskie delete routes indicate success with a 204 HTTP response code.
+         */
+        if (op === 'DELETE' && req.metadata && res.statusCode == 204) {
             var md = req.metadata;
             var owner = md.creator || md.owner;
 
@@ -312,6 +316,13 @@ function auditLogger(options) {
             if (owner) {
                 common.checkAccountSnaplinksEnabled(req, owner,
                     function (enabled) {
+                    /*
+                     * The intent of 'deleted_data_counter' is to count only
+                     * deletes that unlink object backing files from storage
+                     * node filesystems. Directory deletes, zero byte object
+                     * deletes, and snaplink deletes are omitted from this
+                     * count.
+                     */
                     if (md.type === 'object' && md.contentLength > 0) {
                         var storage = md.contentLength * md.sharks.length;
                         labels = {
diff --git a/lib/common.js b/lib/common.js
index 5df530b..b65e6f6 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -337,7 +337,6 @@ function checkAccountSnaplinksEnabled(req, uuid, next) {
     for (var i = 0; i < req.accountsSnaplinksDisabled.length; i++) {
         var account = req.accountsSnaplinksDisabled[i];
         assert.string(account.uuid, 'account.uuid');
-        assert.string(req.caller.account.uuid, 'req.caller.account.uuid');
 
         if (account.uuid === uuid) {
             next(false);
